File: leetcode_762.cpp
class Solution {
public:
    int countPrimeSetBits(int left, int right) {
        unordered_set<int> primes = { 2,3,5,7,11,13,17,19};
        int cnt=0;
        for(int i=left; i<=right; i++){
            int setbit = __builtin_popcount(i);
            if(primes.count(setbit)){
                cnt++;
            }
        }
        return cnt;
    }
};

File: leetcode_762.py
class Solution:
    def isPrime(self,n: int) -> bool:
        if n < 2:
             return False
        for i in range(2,int(n**0.5)+1):
            if(n % i ==0):
                return False
            
        return True
    def countPrimeSetBits(self, left: int, right: int) -> int:
        ans=0
        for i in range(left,right+1):
            setbit = i.bit_count()
            if self.isPrime(setbit):
                ans+=1
            
        return ans
        